File: Client/src/services/savedRecipeService.js
import api from './api';

console.log('[savedRecipeService] loaded, getAll URL:', '/SavedRecipe');

export const savedRecipeService = {
  // SavedRecipe endpoints
  getAll: () => { console.log('[savedRecipeService] getAll called, URL: /SavedRecipe'); return api.get('/SavedRecipe'); },
  getById: (id) => api.get(`/SavedRecipe/${id}`),
  getByCollectionId: (collectionId) => api.get(`/SavedRecipe/collection/${collectionId}`),
  toggle: (data) => api.post('/SavedRecipe/toggle', data),
  create: (data) => api.post('/SavedRecipe', data),
  update: (id, data) => api.put(`/SavedRecipe/${id}`, data),
  delete: (id) => api.delete(`/SavedRecipe/${id}`),

  // Collection endpoints
  getAllCollections: () => api.get('/collection'),
  getCollectionById: (id) => api.get(`/collection/${id}`),
  getDefaultCollection: (accountId) => api.get(`/collection/account/${accountId}/default`),
  createCollection: (data) => api.post('/collection', data),
  updateCollection: (id, data) => api.put(`/collection/${id}`, data),
  deleteCollection: (id) => api.delete(`/collection/${id}`),
};

File: ClientMobile/lib/features/favorites/data/saved_recipe_service.dart
import 'package:smart_meal/core/network/api_client.dart';
import 'package:smart_meal/core/constants/api_constants.dart';

/// Saved recipe / favorites service – mapped from web's savedRecipeService.js
class SavedRecipeService {
  final ApiClient _api = ApiClient();

  Future<List<Map<String, dynamic>>> getAll() async {
    final response = await _api.get(ApiConstants.savedRecipe);
    return (response.data['data'] as List? ?? []).cast<Map<String, dynamic>>();
  }

  Future<Map<String, dynamic>> toggle(Map<String, dynamic> data) async {
    final response = await _api.post(ApiConstants.savedRecipeToggle, data: data);
    return response.data as Map<String, dynamic>;
  }

  Future<Map<String, dynamic>> getDefaultCollection(String accountId) async {
    final response = await _api.get('${ApiConstants.collection}/account/$accountId/default');
    return response.data as Map<String, dynamic>;
  }

  Future<List<Map<String, dynamic>>> getAllCollections() async {
    final response = await _api.get(ApiConstants.collection);
    return (response.data['data'] as List? ?? []).cast<Map<String, dynamic>>();
  }
}
